os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/da_ble_scatternet/ble_tizenrt_scatternet_app.c

```diff
@@ -163,6 +164,40 @@ void ble_tizenrt_scatternet_handle_callback_msg(T_TIZENRT_CLIENT_APP_CALLBACK_MS
             }
         }
 			break;
+
+        case BLE_TIZENRT_CALLBACK_TYPE_CONN:
+        {
+            T_TIZENRT_CONNECTED_CALLBACK_DATA *connected = callback_msg.u.buf;
+            debug_print("\r\n[%s] cb %p conn_id 0x%x conn_type 0x%x addr 0x%x0x%x0x%x0x%x0x%x0x%x", __FUNCTION__,
+                            server_init_parm.connected_cb,
+                            connected->conn_id, connected->conn_type,
+                            connected->remote_bd[0], connected->remote_bd[1], connected->remote_bd[2],
+                            connected->remote_bd[3], connected->remote_bd[4], connected->remote_bd[5]);
+            if(server_init_parm.connected_cb)
+            {
+                da_ble_server_connected_t p_func = server_init_parm.connected_cb;
+                p_func(connected->conn_id, connected->conn_type, connected->remote_bd);
+            } else {
+                debug_print("\r\n[%s] NULL connected callback", __FUNCTION__);
+            }
+            os_mem_free(connected);
+        }
+		    break;
+
+        case BLE_TIZENRT_CALLBACK_TYPE_PROFILE:
+        {
+            T_TIZENRT_PROFILE_CALLBACK_DATA *profile = callback_msg.u.buf;
+            if(profile->cb)
```

Comment:
We'd better check `profile` is NULL or not. It can be dereferencing.

Reply:
I have optimized as you suggest.